### PR TITLE
RxSwift 5.1.1 support

### PR DESCRIPTION
**Pull request by foxware00**

### CODE REVIEW

File: RxASDataSources.podspec [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
      s.requires_arc = true
-     s.swift_version = '5.0'
+     s.swift_versions = '5.0'
 
      s.source_files  = "Sources/**/*.swift"
      s.framework  = "Foundation"
-     s.dependency 'RxSwift', '~> 5.0.0'
-     s.dependency 'RxCocoa', '~> 5.0.0'
+     s.dependency 'RxSwift', '~> 5.1.1'
+     s.dependency 'RxCocoa', '~> 5.1.1'
```
Comment (dangthaison91) [inline RxASDataSources.podspec]: I think version should be 5.1 for more flexible

File: RxASDataSources.podspec [hunk 2]
```diff
@@ -1,21 +1,21 @@
 Pod::Spec.new do |s|
      s.name = 'RxASDataSources'
-     s.version = '0.3.4'
+     s.version = '0.4.2'
```
Comment (dangthaison91) [inline RxASDataSources.podspec]: I think it can be 0.4?

**Review by dangthaison91 — CHANGES_REQUESTED** [review]

**Review by dangthaison91 — APPROVED** [review]

**Comment by foxware00** [comment]
@dangthaison91 Good idea, removed minor versions

**Review by dangthaison91 — APPROVED** [review]
LGTM now, Thanks @foxware00

File: Sources/DataSources/ASCollectionNode+Rx/ASCollectionSectionedDataSource.swift [hunk 3]
```diff
@@ -36,7 +36,8 @@ open class ASCollectionSectionedDataSource<S: SectionModelType>: NSObject, ASCol
 
     fileprivate static func configureSupplementaryViewBlockNotSet(dataSource: ASCollectionSectionedDataSource<S>, node: ASCollectionNode, nodeForSupplementaryElementOfKind kind: String, indexPath: IndexPath) -> ASCellNodeBlock {
         // Users expect collectionNode(_nodeForSupplementaryElementOfKind:at:) will be executed in main thread according to api doc.
-        return { DispatchQueue.main.sync(execute: { dataSource.collectionNode(node, nodeForSupplementaryElementOfKind: kind, at: indexPath) }) }
+        let cellNode = dataSource.collectionNode(node, nodeForSupplementaryElementOfKind: kind, at: indexPath)
```
Comment (dangthaison91) [inline Sources/DataSources/ASCollectionNode+Rx/ASCollectionSectionedDataSource.swift]: Should we keep returning cell block inside `DispatchQueue.main` @foxware00 ?

**Comment by foxware00** [comment]
@dangthaison91 Happy for me to bump to 0.5.0 seeing as the dependency change is breaking?

File: Sources/DataSources/ASCollectionNode+Rx/ASCollectionSectionedDataSource.swift (hunk 3 above)
Comment (foxware00) [reply Sources/DataSources/ASCollectionNode+Rx/ASCollectionSectionedDataSource.swift]: I wasn't sure why this was picked up in this PR, to be honest, but we merged this change a little while ago in #36. It's a valid change and I believe has been in 0.4.1 for a while now.

If you're on the main thread and call `main.sync` it'll deadlock

Comment (foxware00) [reply Sources/DataSources/ASCollectionNode+Rx/ASCollectionSectionedDataSource.swift]: @dangthaison91 looks like it's in `master` so I can't work out why it's coming up here

Comment (foxware00) [reply Sources/DataSources/ASCollectionNode+Rx/ASCollectionSectionedDataSource.swift]: @dangthaison91 update master and it's gone. Also bumped to `0.5.0`

**Comment by dangthaison91** [comment]
@foxware00 I think `0.5` is better as it is aligned with Swift and RxSwift version.

**Comment by foxware00** [comment]
@dangthaison91 Would you be able to make a release in GitHub so others know about the release, please. I'm unable to do so or fetch the changes.

**Comment by dangthaison91** [comment]
Just bumped new pod version @foxware00 :D

```
--------------------------------------------------------------------------------
 🎉  Congrats
 🚀  RxASDataSources (0.5.0) successfully published
 📅  April 16th, 21:24
 🌎  https://cocoapods.org/pods/RxASDataSources
 👍  Tell your friends!
--------------------------------------------------------------------------------
```

**Pull request merged**